### PR TITLE
Fieldlines from vmec

### PR DESCRIPTION
**Pull request by landreman**
This pull request adds the feature of computing field lines from a vmec solution. This involves a 1D root-finding problem to compute theta_pest from theta_vmec. From the result, all the geometric quantities that enter the gyrokinetic equation are computed, such as B cross kappa dot grad psi, |grad alpha|^2, etc. This is essentially a port of the `stella` geometry module. Typical usage:

~~~~.py
    import numpy as np
    from simsopt.mhd.vmec import Vmec
    from simsopt.mhd.vmec_diagnostics import vmec_fieldlines

    v = Vmec('wout_li383_1.4m.nc')                                                                                                       
    theta = np.linspace(-np.pi, np.pi, 50)
    s = 0.5
    alpha = [0, np.pi]                                                                                               
    fl = vmec_fieldlines(v, s, alpha, theta1d=theta)                                                                                       
    print(fl.B_cross_grad_B_dot_grad_alpha)
    print(fl.grad_alpha_dot_grad_alpha)
~~~~

### CODE REVIEW

File: src/simsopt/mhd/vmec_diagnostics.py [hunk 1]
```diff
@@ -651,3 +652,520 @@ def shape_gradient(self):
...
+    # *********************************************************************
+    # Compute the Cartesian components of other quantities we need:
+    # *********************************************************************
+
+    grad_psi_X = grad_s_X * edge_toroidal_flux_over_2pi
+    grad_psi_Y = grad_s_Y * edge_toroidal_flux_over_2pi
+    grad_psi_Z = grad_s_Z * edge_toroidal_flux_over_2pi
+
+    # Form grad alpha = grad (theta_vmec + lambda - iota * phi)
+    phi_center = 0  # Might want to change this in the future
```
Comment (rogeriojorge) [inline src/simsopt/mhd/vmec_diagnostics.py:1036]: Should this be an input for the user?

File: src/simsopt/mhd/vmec_diagnostics.py [hunk 2]
```diff
@@ -651,3 +652,520 @@ def shape_gradient(self):
...
+
+    if plot:
+        import matplotlib.pyplot as plt
+        plt.figure(figsize=(13, 7))
+        nrows = 4
+        ncols = 5
+        variables = ['modB', 'B_sup_theta_vmec', 'B_sup_phi', 'B_cross_grad_B_dot_grad_alpha', 'B_cross_grad_B_dot_grad_psi',
+                     'B_cross_kappa_dot_grad_alpha', 'B_cross_kappa_dot_grad_psi',
+                     'grad_alpha_dot_grad_alpha', 'grad_alpha_dot_grad_psi', 'grad_psi_dot_grad_psi',
+                     'bmag', 'gradpar_theta', 'gradpar_phi', 'gbdrift', 'gbdrift0', 'cvdrift', 'cvdrift0', 'gds2', 'gds21', 'gds22']
```
Comment (rogeriojorge) [inline src/simsopt/mhd/vmec_diagnostics.py:1157]: Is there any way to add a unit test for this plot?

**Review by rogeriojorge — APPROVED** [review]
Looks really good!

File: src/simsopt/mhd/vmec_diagnostics.py (hunk 1 above)
Comment (landreman) [reply src/simsopt/mhd/vmec_diagnostics.py]: Done.

File: src/simsopt/mhd/vmec_diagnostics.py (hunk 2 above)
Comment (landreman) [reply src/simsopt/mhd/vmec_diagnostics.py:1164]: A test for this plotting routing is now added.

**Pull request merged**